pkg/noun/zave.c

```diff
@@ -90,3 +92,13 @@ u3z_uniq(u3_noun som)
     return som;
   }
 }
+
+/* u3z_reap(): promote memoization cache state.
+*/
+void
+u3z_reap(u3p(u3h_root) har_p)
+{
+  // u3h_walk_with(har_p, _cz_merge_memo_cb, &u3R->cax.har_p);
+  // NB *not* u3z_free; _cz_merge_memo_cb() transfers u3z_memo's
+  // u3h_free(har_p);
+}
```

Comment:
For a proof-of-concept, all you have to do here is

```
u3h_uni(u3R->cax.har_p, har_p);
u3h_free(har_p);
```

There's no need to touch bytecode programs.